src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -132,51 +258,16 @@ def perform_delete_previous_version_dataset_transaction(
         previous_version: previous version of the dataset to delete.
         """
 
-        # A stipulation of the @firestore.transactional decorator is the first parameter HAS
-        # to be 'transaction', but since we're using classes the first parameter is always
-        # 'self'. Encapsulating the transaction within this function circumvents the issue.
-        @firestore.transactional
-        def delete_collection_transaction(transaction: firestore.Transaction):
-            previous_version_dataset = (
-                self.datasets_collection.where("survey_id", "==", survey_id)
-                .where("period_id", "==", period_id)
-                .where("sds_dataset_version", "==", previous_version)
-            )
-
-            self._delete_collection(transaction, previous_version_dataset)
-
-        delete_collection_transaction(self.client.transaction())
-
-    def _delete_collection(
-        self,
-        transaction: firestore.Transaction,
-        collection_ref: firestore.CollectionReference,
-    ) -> None:
-        """
-        Recursively deletes the collection and its subcollections.
-
-        Parameters:
-        transaction: the firestore transaction performing the delete.
-        collection_ref: the reference of the collection being deleted.
-        """
-        doc_collection = collection_ref.stream()
-
-        for doc in doc_collection:
-            self._recursively_delete_document_and_sub_collections(
-                transaction, doc.reference
-            )
-
-    def _recursively_delete_document_and_sub_collections(
-        self, transaction: firestore.Transaction, doc_ref: firestore.DocumentReference
-    ) -> None:
-        """
-        Loops through each collection in a document and deletes the collection.
+        previous_version_dataset = (
+            self.datasets_collection.where("survey_id", "==", survey_id)
+            .where("period_id", "==", period_id)
+            .where("sds_dataset_version", "==", previous_version)
+            .stream()
+        )
 
-        Parameters:
-        transaction: the firestore transaction performing the delete.
-        doc_ref: the reference of the document being deleted.
-        """
-        for collection_ref in doc_ref.collections():
-            self._delete_collection(transaction, collection_ref)
+        for dataset in previous_version_dataset:
+            dataset_id = dataset.id
+            logger.info(f"dataset id {dataset_id}")
 
-        transaction.delete(doc_ref)
+        self.delete_collection_in_batches(self.datasets_collection, 100, dataset_id)
```

Comment:
i get an warning that dataset_id may not be set

Reply:
Thanks for picking that up. I have forgotten to change the function call after changing the name. Will fix